BlackCubes/foodtrucks-django-react-native
# Django Setup

## Changes
1. Added a `.gitignore` file, and a settings file for VSCode.
2. Initiated Django and a project.

## Purpose
In order to create a Server, Django needs to be initiated.

## Approach
Before initiating Django, a `.gitignore` file was created at the root level where Django would live (in this case, it would be in the `/server` folder), and this was done so that certain files could not be tracked from GIT.

In the `/server` folder, Django was installed by running the command in the terminal `pip install Django`. Another package was installed called `autopep8` which would be used to automatically format python code. Afterwards, a `requirements.txt` file was created which keeps a record of the versions of all the packages in this project, and so that others could reproduce this project.

Lastly, a Django project was created called `main` (the `/main` folder) which would house the logic of the whole application.

## Testing
1. Pull changes.
2. Open your terminal, go to the `/server` folder, and run `python -m venv env` to create a new virtual environment.
3. On the same `/server` folder in your terminal, activate the virtual environment by running `env/Scripts/activate` on Windows, or `env/bin/activate` on Mac.
4. Then, install the necessary packages for Django by running `python -m pip install -r requirements.txt` or `pip install -r requirements.txt` if the virtual environment is activated.
5. Now, run `python manage.py runserver` in your terminal, go to `http://localhost:8000` on your browser, and make sure the image in **Screenshots** appear.

## Screenshots
![Screenshot 2021-10-01 134844](https://user-images.githubusercontent.com/29642735/135684564-bb5b3775-dea3-446c-906d-7f11d9ee963d.png)


Closes #1
